### PR TITLE
Normalize dtype handling and CSV options

### PR DESCRIPTION
**Pull request by josevalim**
* Remove `names` option in `from_csv` to remove ambiguity
    (users can call rename afterwards)

  * Normalizea and validate `dtypes` in `from_csv`

  * Remove `Inf` which is not a common Elixir value

  * Support returning a list from `Series.transform` to
    unify `dtypes` handling

### CODE REVIEW

File: notebooks/exploring_explorer.livemd [hunk 1]
```diff
@@ -24,19 +24,17 @@ Mix.install([
 
 ## Reading and writing data
 
-Data can be read from delimited files (like CSV), Parquet, and the Arrow IPC (feather) format. You can also load in data from a map or keyword list of columns with `Explorer.DataFrame.new/1`.
+Data can be read from delimited files (like CSV), NDJSon, Parquet, and the Arrow IPC (feather) format. You can also load in data from a map or keyword list of columns with `Explorer.DataFrame.new/1`.
```
Comment (cigrainger) [inline notebooks/exploring_explorer.livemd:27]:
```suggestion
Data can be read from delimited files (like CSV), NDJSON, Parquet, and the Arrow IPC (feather) format. You can also load in data from a map or keyword list of columns with `Explorer.DataFrame.new/1`.
```

File: notebooks/exploring_explorer.livemd [hunk 2]
```diff
@@ -24,19 +24,17 @@ Mix.install([
...
 
 For CSV, your 'usual suspects' of options are available:
 
 * `delimiter` - A single character used to separate fields within a record. (default: `","`)
-* `dtypes` - A keyword list of `[column_name: dtype]`. If `nil`, dtypes are imputed from the first 1000 rows. (default: `nil`)
-* `header?` - Does the file have a header of column names as the first row or not? (default: `true`)
-* `max_rows` - Maximum number of lines to read. (default: `Inf`)
-* `names` - A list of column names. Must match the width of the dataframe. (default: nil)
```
Comment (cigrainger) [inline notebooks/exploring_explorer.livemd:35]: This one is meant for renaming when you read it in or if it doesn't have a header. You could just pipe to `rename` I suppose. It's commonly supported in other languages but I don't think I've just about ever used it.

**Review by cigrainger — COMMENTED** [review]
This is muuuuch tidier.

File: notebooks/exploring_explorer.livemd (hunk 2 above)
Comment (josevalim) [reply notebooks/exploring_explorer.livemd:35]: Yeah. The issue is that it introduces ambiguity for the other option (the names before or after). We would also have to normalize options, etc. I think going with renaming is the cleaner way to do it. :) If necessary, we can add a mention to the docs about rename!

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**